pallets/msa/src/rpc/src/lib.rs

```diff
@@ -21,6 +22,9 @@ pub trait MsaApi<BlockHash, AccountId, BlockNumber> {
 
 	#[rpc(name = "msa_getMsaId")]
 	fn get_msa_id(&self, key: AccountId) -> Result<Option<MessageSenderId>>;
+
+	#[rpc(name = "msa_checkDelegations")]
+	fn check_delegations(&self, delegator_msa_ids: Vec<MessageSenderId>, provider_msa_id: MessageSenderId) -> Result<BTreeMap<MessageSenderId, bool>>;
```

Comment:
I think using a BTreeMap here is a bit overkill. It can just be `Vec` of some custom type or a pair. The reason is that this is a tree and populating it will be `O(NlogN)` but it can just be `O(N).` Maybe this is over optimization but since our resources are limited it might add up.